interface/forms/telemed/view.php

```diff
@@ -0,0 +1,378 @@
+$obj = $formid ? formFetch("form_telemed", $formid) : array();
+
+$query          = "SELECT * FROM patient_data where pid=?";
+$pat_data       =  sqlQuery($query, array($pid));
+
+$query10 = "select  *,form_encounter.date as encounter_date
+               from forms,form_encounter,form_telemed
+               where
+                    forms.deleted != '1'  and
+                    forms.formdir='telemed' and
+                    forms.encounter=form_encounter.encounter  and
+                    forms.form_id=form_telemed.id and
+                    form_telemed.id =? ";
+
+$encounter_data =sqlQuery($query10, array($id));
+$tm_duration = $encounter_data['tm_duration'];
+$tm_subj = $encounter_data['tm_subj'];
+$tm_obj = $encounter_data['tm_obj'];
+$tm_imp = $encounter_data['tm_imp'];
+$tm_plan = $encounter_data['tm_plan'];
+
+$dated = new DateTime($encounter_data['encounter_date']);
+$dated = $dated->format('Y-m-d');
+$visit_date = oeFormatShortDate($dated);
+
+$query = "select * from openemr_postcalendar_events where pc_pid=? and pc_eventDate=?";
+$appt = sqlQuery($query, array($pid,$encounter_data['encounter_date']));
+$reason = $appt['pc_hometext'];
+
+$query          = "SELECT * FROM users where id = ?";
+$prov_data      = sqlQuery($query, array($appt['pc_aid']));
+$id = $form_id;
+
+?><!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/html">
+<head>
+    <link rel="stylesheet" href="<?php echo $webroot; ?>/interface/main/messages/css/reminder_style.css?v=<?php echo $v_js_includes; ?>" type="text/css">
+    <link rel="stylesheet"  href="<?php echo $GLOBALS['web_root']; ?>/library/css/bootstrap_navbar.css?v=<?php echo $v_js_includes; ?>" type="text/css">
+
+    <?php Header::setupHeader(''); ?>
+    <?php $GLOBALS['medex_enable']='';
+        if (($GLOBALS['medex_enable'] == '1')) {
+            $logged_in = $MedEx->login('1');
+            if (empty($_REQUEST['nomenu'])) {
+                $MedEx->display->navigation($logged_in);
+            }
+            //ask MedEx to build a room and send room keys based on my plan (JWT or otherwise);
+            $data['pid']            = $pid;
+            $data['providerID']     = $appt['pc_aid'];
+            $data['eid']            = $appt['pc_eid'];
+            $data['token']          = $logged_in['token'];
+            $data['pat_fname']      = $pat_data['fname'];
+            $data['pat_lname']      = $pat_data['lname'];
+            $data['pat_email']      = $pat_data['email'];
+            $data['pat_mobile']     = $pat_data['mobile'];
+            $data['pat_allowemail'] = $pat_data['hipaa_allowemail'];
+            $data['pat_allowsms']   = $pat_data['hipaa_allowsms'];
+    
+            $MedEx->curl->setUrl($MedEx->getUrl('custom/TM_bot'));
+            $MedEx->curl->setData($data);
+            $MedEx->curl->makeRequest();
+            $TM = $MedEx->curl->getResponse();
+        } else {
+            $token = openssl_random_pseudo_bytes(4);
+            $token = bin2hex($token);
```

Comment:
For the token, how many characters do you need to make here and what characters are ok to include? With that info, I can direct you how to make a crypto secure randomly generated token using a quick function call to a cool random token generator that OpenEMR has (openssl_random_pseudo_bytes is not crypto secure).

Reply:
The goal is to create a unique room.  IT does not need to be crypto secure but I am curious about where this function is in openemr...

Reply:
Do this instead.
1. Place following use at top:
```
use OpenEMR\Common\Utils\RandomGenUtils;
```
2. Then can create a truly random token via:
```
$token = RandomGenUtils::produceRandomString(8, "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789");
```
(note the 1st parameter is the number of character and the 2nd parameters is which characters to choose from)

Reply:
For example, if truly only wanted 8 hex characters (which I doubt is what you are looking for), then would use "0123456789ABCDEF" for the second parameters.